packages/utils/src/bytes.ts

```diff
@@ -63,8 +63,14 @@ const rootBuf = Buffer.alloc(32);
  * Convert a Uint8Array, length 32, to 0x-prefixed hex string
  */
 export function toRootHex(root: Uint8Array): string {
-  rootBuf.set(root);
-  return `0x${rootBuf.toString("hex")}`;
+  if (root.length === 32) {
+    rootBuf.set(root);
+    return `0x${rootBuf.toString("hex")}`;
+  }
+
+  // this is to compliant to toHex() or toHexString() behavior
+  // in case unit tests, or consumers pass in a Uint8Array that is not 32 bytes accidentally
+  return toHex(root);
```

Comment:
@wemeetagain @nflaig I want to make it compliant to the old apis so we don't have to worry about the wrong usage

Reply:
Are calling this in any unit tests? since performance is not relevant there we might be fine with just throwing an error if byte length is invalid. So either the test values have to be adapted or can use `toHex` there instead

Reply:
agree to throw error here if it's not 32
my other thought is should we append `"0x"` here? we mostly use it for internal track and log
the down side with appending `"0x"` is almost 40% more memory allocation https://github.com/ChainSafe/lodestar/blob/44b215667efb60366f28f7b11c97c7c092a3dffc/packages/beacon-node/test/memory/bytesHex.ts#L10

Reply:
don't we have `toMemoryEfficientHexStr` for that? https://github.com/ChainSafe/lodestar/blob/44b215667efb60366f28f7b11c97c7c092a3dffc/packages/state-transition/src/cache/pubkeyCache.ts#L16

I would like to have hex values prefixed with 0x in logs and we also use this on the api which needs 0x-prefix on all hex strings.

Reply:
ok then let's just keep `0x`

Reply:
aren't most / all of these hex string transient anyways?